mathesar_ui/src/component-library/label/Label.svelte

```diff
@@ -3,18 +3,36 @@
     LabelController,
     setLabelControllerInContext,
   } from './LabelController';
+  import { getLabelIdFromInputId } from './labelUtils';
 
   export let controller = new LabelController();
 
   $: setLabelControllerInContext(controller);
   $: ({ inputId, disabled } = controller);
+
+  function handleClick() {
+    const inputElement = document.querySelector(`#${$inputId}`) as
+      | HTMLInputElement
+      | undefined;
+    // Why do we imperatively focus the input element when the DOM performs this
+    // behavior natively? Because we might need to associate a label with a
+    // custom input component that doesn't have that doesn't contain any native
+    // DOM elements with that behavior.
+    inputElement?.focus();
+  }
 </script>
 
 <!--
   TODO use a better class name. We already had some "label" classes in use. We
   need to figure out how to avoid naming collisions here.
 -->
-<label for={$inputId} class="label-component" class:disabled={$disabled}>
+<label
+  for={$inputId}
+  class="label-component"
+  class:disabled={$disabled}
+  id={getLabelIdFromInputId($inputId)}
```

Comment:
just curios what is the use of giving the `id` prop to the label tag?

Reply:
In the record page, I wanted a way to semantically associate the LinkedRecordInput elements (which are not actual `<input>` elements, but rather `<span>` elements with `role="listbox"`) with their corresponding form labels. Associating the label with the input improves a11y and also allow users to focus the input by clicking on the label. Because LinkedRecordInput is a fake input, I need to resort to using aria attributes instead of the standards approaches. So in `LinkedRecordInput.svelte`, I'm putting `aria-labelledby={getLabelIdFromInputId(id)}` on the `<span>`. At the other end, `Label.svelte`  sets `id={getLabelIdFromInputId($inputId)}` on the `<label>` element and the two id values match up, associating the label with the input. This was the best way I could find to do that. If you have better way in-mind, I'm open to changing this.